scrapers_next/hi/people.py

```diff
@@ -1,89 +1,128 @@
-        return f"FormSource('{self.url}', '{self.form_xpath}', '{self.button_label}')"
+    def process_item(self, item):
+        a_tag = item.xpath("a")[0]
+        member_page = a_tag.get("href")
+        member_photo = a_tag.xpath("img")[0].get("src")
 
+        member_text = a_tag.text_content().strip()
 
-class Legislators(HtmlListPage):
-    source = FormSource(
-        "https://www.capitol.hawaii.gov/members/legislators.aspx", "//form", "Show All"
-    )
-    selector = CSS("#ctl00_ContentPlaceHolderCol1_GridView1 tr")
-
-    LABELS = {
-        "first_name": "LabelFirst",
-        "party": "LabelParty",
-        "room": "LabelRoom2",
-        "voice": "LabelPhone2",
-        "fax": "LabelFAX2",
-        "email": "HyperLinkEmail",
-        "chamber": "LabelDis",
-        "district": "LabelDistrict",
-    }
+        # Conditional re-formats member_text in cases with multiple commas
+        #   Ex: Richards\r\n, III, Herbert M. "Tim" (D)
+        multi_commas = self.multi_commas_regex.search(member_text)
+        if multi_commas:
+            single_line = member_text.replace("\r\n", "")
+            comma_split = [x.strip() for x in single_line.split(",")]
+            member_text = f"{' '.join(comma_split[0:2])}, {comma_split[-1]}"
 
-    def process_item(self, item):
-        try:
-            link = CSS("a").match(item)[1]
-        except SelectorError:
-            self.skip()
-        data = {
-            "last_name": link.text_content(),
-            "url": link.get("href"),
-        }
-        for key, label in self.LABELS.items():
-            data[key] = CSS(f"[id$={label}]").match_one(item).text_content().strip()
-
-        party = {"(D)": "Democratic", "(R)": "Republican"}[data["party"]]
-        address = "Hawaii State Capitol, Room " + data["room"]
-        chamber = "upper" if data["chamber"] == "S" else "lower"
+        leader = self.leader_regex.search(member_text)
+        if leader:
+            name_parts = [x.strip() for x in leader.groups()]
+            # Extracts title: only leaders have one listed
+            title = name_parts.pop()
+        else:
+            regular_member = self.regular_regex.search(member_text)
+            name_parts = [x.strip() for x in regular_member.groups()]
+
+        # Extracts party abbreviation from list
+        party = name_parts.pop()
+
+        last_name, first_name = name_parts[0], " ".join(name_parts[1:])
+
+        dist_text = item.xpath("div/a")[0].text_content().strip()
+        chamber, district = [x.strip() for x in dist_text.split("District")]
+
+        contact_info = item.xpath("div/address")[0]
+        split_lines = contact_info.text_content().split("\r\n")
+        contact_list = [x.strip() for x in split_lines if len(x.strip())]
+
+        state_addr_base = "415 S Beretania St, Honolulu, HI 96813"
+        capitol_addr = f"{contact_list[0]}, {state_addr_base}"
+
+        cap_phone, cap_fax = [x.split(":")[-1].strip() for x in contact_list]
```

Comment:
I would recommend splitting this out to be specific for `contact_list[1] & [2]`, since it's expecting 2 values but `contact_list` is usually 4

Reply:
So glad you caught that. I had it written into two separate expressions but accidentally forgot isolating it to just those two elements of `contact_list` when I converted this into a list comprehension.

I can rewrite with this simple fix:
`cap_phone, cap_fax = [x.split(":")[-1].strip() for x in contact_list[1:3]`

Reply:
Resolved in [newest commit](https://github.com/openstates/openstates-scrapers/pull/4247/commits/399cb4b2a6dff5f1160c4b686f77ecebc49576e1)